sentry-apollo-3/src/main/java/io/sentry/apollo3/SentryApollo3HttpInterceptor.kt

```diff
@@ -46,69 +47,64 @@ class SentryApollo3HttpInterceptor(private val hub: IHub = HubAdapter.getInstanc
             }
         }
     }
 
-    private fun Long?.ifHasValidLength(fn: (Long) -> Unit) {
-        if (this != null && this != -1L) {
-            fn.invoke(this)
-        }
+    private fun removeSentryInternalHeaders(headers: List<HttpHeader>): List<HttpHeader> {
+        return headers.filterNot { it.name == SENTRY_APOLLO_3_VARIABLES || it.name == SENTRY_APOLLO_3_OPERATION_NAME || it.name == SENTRY_APOLLO_3_OPERATION_TYPE }
     }
 
     private fun startChild(request: HttpRequest, activeSpan: ISpan): ISpan {
-        val buffer = Buffer()
-        request.body?.writeTo(buffer)
-
-        val reader = InputStreamReader(buffer.inputStream())
-
-        val content = try {
-            hub.options.serializer.deserialize(
-                reader,
-                ApolloRequestBodyContent::class.java,
-                ApolloRequestBodyContent.Deserializer
-            )
-        } catch (e: Throwable) {
-            hub.options.logger.log(SentryLevel.ERROR, "Error deserializing Apollo Request Body.", e)
-        }
-
-        val span = when (content) {
-            is ApolloRequestBodyContent -> createSpanFromBodyContent(activeSpan, request, content)
-            else -> activeSpan.startChild("apollo.request.unknown")
-        }
-
-        val operationId = request.headers.firstOrNull { it.name == "X-APOLLO-OPERATION-ID" }?.value ?: "unknown"
+        val url = request.url
+        val method = request.method
+
+        val operationName = operationNameFromHeaders(request)
+        val operation = operationName ?: "apollo.client"
+        val operationType = request.valueForHeader(SENTRY_APOLLO_3_OPERATION_TYPE) ?: method
+        val operationId = request.valueForHeader("X-APOLLO-OPERATION-ID")
+        val variables = request.valueForHeader(SENTRY_APOLLO_3_VARIABLES)
+        val description = "$operationType ${operationName ?: url}"
+
+        return activeSpan.startChild(operation, description).apply {
+            operationId?.let {
+                setData("operationId", it)
+            }
 
-        return span.apply {
-            setData("operationId", operationId)
+            variables?.let {
+                setData("variables", it)
+            }
         }
     }
 
-    private fun createSpanFromBodyContent(activeSpan: ISpan, request: HttpRequest, content: ApolloRequestBodyContent): ISpan {
-        val operationType = parseOperationType(content)
-        val operationName = content.operationName ?: request.headers.firstOrNull { it.name == "X-APOLLO-OPERATION-NAME" }?.value ?: "unknown"
-        val description = "$operationType $operationName"
-
-        return activeSpan.startChild(operationName, description).apply {
-            setData("variables", content.variables.toString())
-        }
+    private fun operationNameFromHeaders(request: HttpRequest): String? {
+        return request.valueForHeader(SENTRY_APOLLO_3_OPERATION_NAME) ?: request.valueForHeader("X-APOLLO-OPERATION-NAME")
```

Comment:
Can we add a comment somewhere to document that apollo headers are not present for batched operations.

Reply:
`RequestHeaders` are present on batched operations as well. However, the `ResponseHeaders` are stripped in batched operations.

Working around it by trying the header first and if that doesn't have a valid size, I'm querying the buffer.

```
            httpResponse.headersContentLength().ifHasValidLength { contentLength ->
                breadcrumb.setData("response_body_size", contentLength)
            }

            if (!breadcrumb.data.containsKey("response_body_size")) {
                httpResponse.body?.buffer?.size?.ifHasValidLength { contentLength ->
                    breadcrumb.setData("response_body_size", contentLength)
                }
            }
```